### PR TITLE
Cot testcase command

### PR DESCRIPTION
**Pull request by Neketek**
The first version of ```cot-testcase``` command.
It has next features:
1. Filename. Required parameter. Syntax ```--file /path/to/fle```. Accepted formats: json, yaml. 
1. cfn-lint integration. Without config. Can be turned off by ```--no-lint``` flag.
1. cfn-nag integration. Without config. Can be turned off by ```--no-vulnerability-check``` flag. 
1. Match check. Syntax ```--match <path.in.json> <valid json value>```
1. Resource check. Syntax ```--resource <id> <type>```
1. Output check. Syntax ```--output <id>```
1. Length check. Syntax ```--length <path.in.json> <int>```
1. Exists check. Syntax ```--exists <path.in.json>``` Null values considered existing. 
1. Not empty check. Syntax ```--not-empty <path.in.json>``` Null values considered empty. Can check length of dicts, strings and lists.

NOTE:
All checks can be used multiple times. 
The path inside arrays uses **.** instead of the **[]** selector.  Example: ```Template.Array.0.Value```

Example:
https://github.com/codeontap/gen3-cli/pull/15/files#diff-e0b1cf7133419eea824a5b3d27dcc072

### CODE REVIEW

File: gen3-cli/cot-testcase.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#! /bin/bash
```
Comment (koriaf) [inline gen3-cli/cot-testcase.sh]: although this space after the `!` character could work in some cases I believe it should not be there

File: gen3-cli/tests/system/cmdb/alm/conftest.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
+from cot.backend.generate.cmdb import account
+
+
+conf = dict(
+    cmdb=dict(
+        tenant=dict(
+            tenant_id='tenant_id',
+            tenant_name='tenant',
+            domain_stem='domain.stem.com',
```
Comment (koriaf) [inline gen3-cli/tests/system/cmdb/alm/conftest.py]: I'd rather use some domain owned by us for demo purpose - Michael could advise

Comment (roleyfoley) [reply gen3-cli/tests/system/cmdb/alm/conftest.py]: codeontap.io would work, we own it and it’s our product

Comment (Neketek) [reply gen3-cli/tests/system/cmdb/alm/conftest.py]: Ok, no problem.

**Pull request merged**